src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -0,0 +1,907 @@
+			warnx("[lpe] lidar init: alt %d cm",
+					int(100*_lidarAltHome));
+			_lidarInitialized = true;
+		}
+	}
+}
+
+void BlockLocalPositionEstimator::initFlow() {
+	// collect flow data
+	if (!_flowInitialized) {
+		// increament sums for mean
+		_flowAltHome += _sub_flow.ground_distance_m;
+		if (_flowInitCount++ > 200) {
+			_flowAltHome /= _flowInitCount;
+			mavlink_log_info(_mavlink_fd, "[lpe] flow init: "
+					"alt %d cm",
+					int(100*_flowAltHome));
+			warnx("[lpe] flow init: alt %d cm",
+					int(100*_flowAltHome));
+			_flowInitialized = true;
+		}
+	}
+}
+
+void BlockLocalPositionEstimator::initVision() {
+	// collect vision data
+	if (!_visionInitialized) {
+		// increament sums for mean
+		math::Vector<3> pos;
+		pos(0) = _sub_vision.x;
+		pos(1) = _sub_vision.y;
+		pos(2) = _sub_vision.z;
+		_visionHome += pos;
+		if (_viconInitCount++ > 200) {
+			_visionHome /= _visionInitCount;
+			mavlink_log_info(_mavlink_fd, "[lpe] vision init: "
+					"%f, %f, %f m", double(pos(0)), double(pos(1)), double(pos(2)));
+			warnx("[lpe] vision init: "
+					"%f, %f, %f m", double(pos(0)), double(pos(1)), double(pos(2)));
+			_visionInitialized = true;
+		}
+	}
+}
+
+void BlockLocalPositionEstimator::initVicon() {
+	// collect vicon data
+	if (!_viconInitialized) {
+		// increament sums for mean
+		math::Vector<3> pos;
+		pos(0) = _sub_vicon.x;
+		pos(1) = _sub_vicon.y;
+		pos(2) = _sub_vicon.z;
+		_viconHome += pos;
+		if (_viconInitCount++ > 200) {
+			_viconHome /= _viconInitCount;
+			mavlink_log_info(_mavlink_fd, "[lpe] vicon init: "
+					"%f, %f, %f m", double(pos(0)), double(pos(1)), double(pos(2)));
+			warnx("[lpe] vicon init: "
+					"%f, %f, %f m", double(pos(0)), double(pos(1)), double(pos(2)));
+			_viconInitialized = true;
+		}
+	}
+}
+
+void BlockLocalPositionEstimator::publishLocalPos() {
+	// publish local position
+	if (isfinite(_x(X_x)) && isfinite(_x(X_y)) && isfinite(_x(X_z)) &&
+		isfinite(_x(X_vx)) && isfinite(_x(X_vy))
+		&& isfinite(_x(X_vz))) {
+		_pub_lpos.timestamp = _timeStamp;
+		_pub_lpos.xy_valid = true;
+		_pub_lpos.z_valid = true;
+		_pub_lpos.v_xy_valid = true;
+		_pub_lpos.v_z_valid = true;
+		_pub_lpos.x = _x(X_x);  // north
+		_pub_lpos.y = _x(X_y);  // east
+		_pub_lpos.z = _x(X_z); // down
+		_pub_lpos.vx = _x(X_vx);  // north
+		_pub_lpos.vy = _x(X_vy);  // east
+		_pub_lpos.vz = _x(X_vz); // down
+		_pub_lpos.yaw = _sub_att.yaw;
+		_pub_lpos.xy_global = _gpsInitialized;
+		_pub_lpos.z_global = _baroInitialized;
+		_pub_lpos.ref_timestamp = _sub_home.timestamp;
+		_pub_lpos.ref_lat = _map_ref.lat_rad*180/M_PI;
+		_pub_lpos.ref_lon = _map_ref.lon_rad*180/M_PI;
+		_pub_lpos.ref_alt = _sub_home.alt;
+		// TODO, terrain alt
+		_pub_lpos.dist_bottom = -_x(X_z);
+		_pub_lpos.dist_bottom_rate = -_x(X_vz);
+		_pub_lpos.surface_bottom_timestamp = 0;
+		_pub_lpos.dist_bottom_valid = true;
+		_pub_lpos.eph = sqrtf(_P(X_x, X_x) + _P(X_y, X_y));
+		_pub_lpos.epv = sqrtf(_P(X_z, X_z));
+		_pub_lpos.update();
```

Comment:
Is this supposed to be something more like

``` c++
if(_pub_lpos > 0) {
  orb_publish(ORB_ID(vehicle_local_position), _pub_lpos, &local_pos);
} else {
  _pub_lpos = orb_advertise(ORB_ID(vehicle_local_position), &local_pos);
}
```

Where `local_pos` is a variable contained within the `BlockLocalPositionEstimator` object. Just to be clear, this is a question as it's structure I've seen elsewhere in the code such as `ekf_att_pos_estimator_main.cpp`. I'm asking because I tried testing this but the commander app refused to let me into posctl mode.